source/grid/grid_tools.cc

```diff
@@ -2405,6 +2405,36 @@ namespace GridTools
                            Triangulation<dim,spacedim> &triangulation,
                            const SparsityTools::Partitioner  partitioner
                           )
+  {
+    std::vector<unsigned int> cell_weights;
+
+    // Get cell weighting if a signal has been attached to the triangulation
+    if (!triangulation.signals.cell_weight.empty())
+      {
+        cell_weights = std::vector<unsigned int> (triangulation.n_active_cells(), std::numeric_limits<unsigned int>::max());
```

Comment:
-> `cell_weights.resize (..., ...);`

Reply:
Thanks, I wasn't aware that `resize` with a default value existed.